### ISSUE TITLE
Child class with template parameter named differently than parent: unqualified base class in tagfile

### ISSUE THREAD

**Issue by cgrooves** [doxygen/doxygen#10499]
**Describe the bug**
The setup is as follows:
- A base class with a template parameter, inside of a namespace
- A child class with a template parameter that passes it through to the base class template, only name the template parameter differently
- Export the tagfile

Example:

```cpp
namespace parent
{
/// \brief Some Parent class
template<typename T>
struct Parent
{
    T foo_ = 0; ///< the parent's foo
};
}

namespace child
{
/// \brief Some derived class
template<typename FooType>
struct Child
    : public parent::Parent<FooType>
{
    FooType bar_ = 0; ///< the child's bar
};
}
```
 
**Expected behavior**
In the tagfile, I would expect to see the namespace-qualified base class listed with its template parameter. Instead, the base class is _not_ fully qualified.

Expected:
```
  <compound kind="struct">
    <name>child::Child</name>
    <filename>structchild_1_1Child.html</filename>
    <templarg>typename FooType</templarg>
    <base>parent::Parent</base>
```

Produced:
```
  <compound kind="struct">
    <name>child::Child</name>
    <filename>structchild_1_1Child.html</filename>
    <templarg>typename FooType</templarg>
    <base>Parent&lt; FooType &gt;</base>
```

This creates further complications if the Child and the Parent class are named the same, since any project that consumes this one's tagfile will produce an error about a possible recursive relationship between the parent and child class. Change the above example to where both the `Child` and `Parent` class are renamed to `FooClass`, and create another Doxygen project that uses this one's tagfile, and you will get the following error:

```
<unknown>:3: warning: Detected potential recursive class relation between class child::FooClass and base class FooClass< FooType >!
```

**To Reproduce**
See the README.md in the attached .zip. Will produce the FooProj.tags file with the unqualified base class, and show the "recursive class relation" when building the BarProj project documentation.

[doxygenExample.zip](https://github.com/doxygen/doxygen/files/13721204/doxygenExample.zip)

**Version**
Doxygen 1.9.6, RHEL8 and Windows 10.

**Comment by doxygen** [comment]
@cgrooves Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
